### ISSUE TITLE
a11y: select state not defined for buttonGroup

### ISSUE THREAD

**Issue by aalves08** [rancher/dashboard#13814]
a11y issue 74

**Setup**
- Rancher version:
- Rancher UI Extensions:
- Browser type & version:

**Describe the bug**

The selected state is not programmatically defined for the buttons (ButtonGroup) resulting in screen reader users finding it difficult to understand the button's selected state. 

Main Content
 - Theme Section
  Light button, Auto button, Dark button 
 - YAML Editor Key Mapping section
  Normal human button, Emacs button, Vim button
 - Helm Charts
  Show Releases Only, Include Prerelease Versions

Add `aria-pressed ="true"` attribute when the button is in the selected state. Ensure that the value of "aria-pressed" attribute is updated on user interaction via scripting.

**To Reproduce**

**Result**
The selected state is not programmatically defined for the buttons (ButtonGroup) resulting in screen reader users finding it difficult to understand the button's selected state. 

**Expected Result**

The screen reader should announce the selected state for the button.

**Screenshots**

**Additional context**

**Comment by a-arias** [comment]
**TESTS** 
Rancher : v2.12-733fda1f8d80127b7afb834c938cb754dc8e1894-head
UI: master https://github.com/rancher/dashboard/commit/ac02757be26ecca167a9dfb26354c1defcc0d7a7
Fix verified ✅ 

<img width="1562" alt="Image" src="https://github.com/user-attachments/assets/486243eb-35db-44d2-a20d-71c632750d73" />
<img width="1562" alt="Image" src="https://github.com/user-attachments/assets/3441c343-1add-4a4c-acac-2dc930e43b27" />
<img width="1562" alt="Image" src="https://github.com/user-attachments/assets/d469a3a4-4a5d-494f-a85c-c80b4c960258" />


https://github.com/user-attachments/assets/dda8fe1a-61aa-4b87-a4eb-79c207779ba0
